src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlBinaryStringTranslator.cs

```diff
@@ -0,0 +1,153 @@
+// DOCUMENTATION, EVEN IF THE NPGSQL DEVELOPMENT TEAM HAS BEEN ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+// THE NPGSQL DEVELOPMENT TEAM SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
+// ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
+// TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+
+#endregion
+
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Security.Cryptography;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Query.Expressions;
+using Microsoft.EntityFrameworkCore.Query.ExpressionTranslators;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.ExpressionTranslators.Internal
+{
+    /// <summary>
+    /// Provides translation services for PostgreSQL binary string functions and operators.
+    /// </summary>
+    /// <remarks>
+    /// See: https://www.postgresql.org/docs/current/static/functions-binarystring.html
+    /// </remarks>
+    public class NpgsqlBinaryStringTranslator : IMethodCallTranslator
+    {
+        #region MethodInfo
+
+        [NotNull] static readonly MethodInfo ComputeHash1 =
+            typeof(HashAlgorithm).GetRuntimeMethod(nameof(HashAlgorithm.ComputeHash), new[] { typeof(byte[]) });
+
+        [NotNull] [ItemNotNull] static readonly MethodInfo[] SupportedCreateMethods =
+        {
+            typeof(MD5).GetRuntimeMethod(nameof(MD5.Create), Type.EmptyTypes),
+            typeof(SHA256).GetRuntimeMethod(nameof(SHA256.Create), Type.EmptyTypes),
+            typeof(SHA384).GetRuntimeMethod(nameof(SHA384.Create), Type.EmptyTypes),
+            typeof(SHA512).GetRuntimeMethod(nameof(SHA512.Create), Type.EmptyTypes),
+        };
+
+        #endregion
+
+        /// <summary>
+        /// The backend version to target.
+        /// </summary>
+        [CanBeNull] readonly Version _postgresVersion;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NpgsqlBinaryStringTranslator"/> class.
+        /// </summary>
+        /// <param name="postgresVersion">The backend version to target.</param>
+        public NpgsqlBinaryStringTranslator([CanBeNull] Version postgresVersion)
+            => _postgresVersion = postgresVersion;
+
+        /// <inheritdoc />
+        [CanBeNull]
+        public Expression Translate(MethodCallExpression e)
+        {
+            // TODO: disable when version is unspecified until 11.0 is released.
+            if (_postgresVersion == null)
+                return null;
+
+            if (SupportedCreateMethods.Contains(e.Method))
+                return e;
+
+            return TranslateComputeHash1(e) ?? TranslateExtensions(e);
+        }
+
+        [CanBeNull]
+        Expression TranslateComputeHash1([NotNull] MethodCallExpression e)
```

Comment:
Why is 1 at the end?

Reply:
There are other overloads of ComputeHash (with array sunsetting). I just hadn't written their translations yet.